atc/syslog/drainer.go

```diff
@@ -49,51 +50,66 @@ func (d *drainer) Run(ctx context.Context) error {
 			logger.Error("Syslog drainer connecting to server error.", err)
 			return err
 		}
-		defer syslog.Close()
+
+		// ignore any errors coming from syslog.Close()
+		defer db.Close(syslog)
 
 		for _, build := range builds {
-			events, err := build.Events(0)
+			err := d.drainBuild(logger, build, syslog)
 			if err != nil {
-				logger.Error("Syslog drainer getting build events error.", err)
 				return err
 			}
+		}
+	}
+	return nil
+}
 
-			for {
-				ev, err := events.Next()
-				if err != nil {
-					if err == db.ErrEndOfBuildEventStream {
-						break
-					}
-					logger.Error("Syslog drainer getting next event error.", err)
-					return err
-				}
-
-				if ev.Event == "log" {
-					var log event.Log
-
-					err := json.Unmarshal(*ev.Data, &log)
-					if err != nil {
-						logger.Error("Syslog drainer unmarshalling log error.", err)
-						return err
-					}
-
-					payload := log.Payload
-					tag := build.TeamName() + "/" + build.PipelineName() + "/" + build.JobName() + "/" + build.Name() + "/" + string(log.Origin.ID)
-
-					err = syslog.Write(d.hostname, tag, time.Unix(log.Time, 0), payload)
-					if err != nil {
-						logger.Error("Syslog drainer sending to server error.", err)
-						return err
-					}
-				}
+func (d *drainer) drainBuild(logger lager.Logger, build db.Build, syslog *Syslog) error {
+	events, err := build.Events(0)
+	if err != nil {
+		logger.Error("Syslog drainer getting build events error.", err)
```

Comment:
Hey, maybe instead of directly using the `logger` that gets passed as an argument, we'd be better of starting a session before so that one can more quickly jump to where the error log happens.

Reply:
The `logger` here is from https://github.com/concourse/concourse/blob/1f56328810f3b8e7f912b4da289550a231eba045/atc/syslog/drainer.go#L39
`drainBuild` is only used internally at https://github.com/concourse/concourse/blob/1f56328810f3b8e7f912b4da289550a231eba045/atc/syslog/drainer.go#L58

Since the logger is dependent on the context of the runner, I don't think we should store it in the `drainer struct` and the only option left is to pass it in as an arg

wdyt?